File: sdk/java/src/main/java/io/agentsafe/spl/Verifier.java
package io.agentsafe.spl;

/**
 * Top-level SPL policy verification.
 */
public final class Verifier {
    private Verifier() {}

    public record Result(boolean allow) {}

    /**
     * Evaluate an SPL policy AST against a request within an environment.
     * @throws SplException if the token is sealed
     */
    public static Result verify(Node policyAst, Env env) {
        if (env.sealed) {
            throw new SplException("token is sealed and cannot be attenuated");
        }
        Evaluator evaluator = new Evaluator(env);
        Node result = evaluator.eval(policyAst);
        return new Result(result.isTruthy());
    }
}

File: sdk/csharp/AgentSafe.Spl/Verifier.cs
namespace AgentSafe.Spl;

public static class Verifier
{
    public record VerifyResult(bool Allow);

    public static VerifyResult Verify(Node policyAst, Env env)
    {
        var evaluator = new Evaluator(env);
        var result = evaluator.Eval(policyAst);
        return new VerifyResult(result.IsTruthy());
    }
}
